src/scipp/ndimage/__init__.py

```diff
@@ -0,0 +1,247 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+"""Sub-package for multidimensional image processing.
+
+This subpackage provides wrappers for a subset of functions from
+:py:mod:`scipy.ndimage`.
+"""
+from functools import wraps
+from typing import Callable, Dict, Union
+
+import scipy.ndimage
+
+from ..core import Variable, DataArray
+from ..core import CoordError, DimensionError, VariancesError
+from ..core import empty_like, islinspace, ones
+
+
+def _ndfilter(func: Callable) -> Callable:
+
+    @wraps(func)
+    def function(x: Union[Variable, DataArray], **kwargs) -> Union[Variable, DataArray]:
+        if 'output' in kwargs:
+            raise TypeError("The 'output' argument is not supported")
+        if x.variances is not None:
+            raise VariancesError(
+                "Filter cannot be applied to input array with variances.")
+        if getattr(x, 'masks', None):
+            raise ValueError("Filter cannot be applied to input array with masks.")
+        return func(x, **kwargs)
+
+    return function
+
+
+def _delta_to_positional(x: Union[Variable, DataArray], dim, index, dtype):
+    if not isinstance(index, Variable):
+        return index
+    coord = x.coords[dim]
+    if not islinspace(coord, dim).value:
+        raise CoordError(
+            f"Data points not regularly spaced along {dim}. To ignore this, "
+            f"provide a plain value (convertible to {dtype.__name__}) instead of a "
```

Comment:
I don't think this works:
```.py
>>> sc.DType.float64.__name__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'scipp._scipp.core.DType' object has no attribute '__name__'
```
Why did you not use `{dtype}`?

Or is `dtype` only allowed to be `int` and `float`? (Looking at `dtype(pos)` below)

Reply:
`dtype` is only `int` or `float`, yes. `{dtype}` renders to something like `"<class 'int'>"`, so it is less user readable.

Reply:
ok